spring-xd-codec/src/main/java/org/springframework/xd/dirt/integration/bus/serializer/kryo/CompositeKryoRegistrar.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2015 the original author or authors.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.xd.dirt.integration.bus.serializer.kryo;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.esotericsoftware.kryo.Registration;
+
+import org.springframework.util.Assert;
+import org.springframework.util.CollectionUtils;
+import org.springframework.xd.dirt.integration.bus.serializer.SerializationException;
+
+/**
+ * A {@link org.springframework.xd.dirt.integration.bus.serializer.kryo.KryoRegistrar} that delegates and validates
+ * registrations across all components.
+ * @author David Turanski
+ * @since 1.2
+ */
+public class CompositeKryoRegistrar extends AbstractKryoRegistrar {
+
+	private final List<AbstractKryoRegistrar> delegates;
```

Comment:
See if can refactor to use a modified KryoRegistrar interface vs. abstract base class.